Std/Control/Lemmas.lean

```diff
@@ -0,0 +1,25 @@
+import Std.Tactic.Ext
+
+namespace ReaderT
+
+attribute [ext] ReaderT.ext
+
+@[simp] theorem run_failure [Monad m] [Alternative m] (ctx : ρ) :
+  (failure : ReaderT ρ m α).run ctx = failure := rfl
```

Comment:
As far as I can tell the naming conventions as [written](https://leanprover-community.github.io/contribute/naming.html) are silent on whether this should be `run_failure` or `failure_run`.

I think we should start becoming more consistent about this, and really hope that we can go with `failure_run` here:
* We all use dot notation, and basic principle is "important fragments of the statement, read left to right".
* This is obviously a theorem about `failure` more than it is a theorem about `run`, so `failure` deserves top billing.

Obviously we should not hold up this PR for such a discussion, of course. :-)

Reply:
@digama0 @joehendrix

Reply:
The [last time](https://github.com/leanprover/std4/pull/329#issuecomment-1786193983) this came up, we ended up with "mostly prefix, but binary operators are infix" for a not-completely-defined notion of what constitutes a binary operator. Here `run` could conceivably be interpreted as a binary operator, but I lean toward not doing so because it doesn't have an associated notation.